src/main/java/org/jetbrains/plugins/innerbuilder/InnerBuilderGenerator.java

```diff
@@ -0,0 +1,557 @@
+package org.jetbrains.plugins.innerbuilder;
+
+import static org.jetbrains.plugins.innerbuilder.InnerBuilderUtils.areTypesPresentableEqual;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+
+import org.jetbrains.annotations.NonNls;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import com.intellij.codeInsight.generation.PsiFieldMember;
+
+import com.intellij.ide.util.PropertiesComponent;
+
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.project.Project;
+
+import com.intellij.psi.JavaPsiFacade;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiCodeBlock;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiElementFactory;
+import com.intellij.psi.PsiField;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiMethod;
+import com.intellij.psi.PsiModifier;
+import com.intellij.psi.PsiModifierList;
+import com.intellij.psi.PsiParameter;
+import com.intellij.psi.PsiPrimitiveType;
+import com.intellij.psi.PsiStatement;
+import com.intellij.psi.PsiType;
+import com.intellij.psi.codeStyle.CodeStyleManager;
+import com.intellij.psi.codeStyle.JavaCodeStyleManager;
+import com.intellij.psi.util.PropertyUtil;
+import com.intellij.psi.util.PsiUtil;
+
+public class InnerBuilderGenerator implements Runnable {
+    @NonNls
+    private static final String BUILDER_CLASS_NAME = "Builder";
+    @NonNls
+    private static final String JSR305_NONNULL = "javax.annotation.Nonnull";
```

Comment:
This seems somehow wrong. http://types.cs.washington.edu/checker-framework/current/api/org/checkerframework/checker/nullness/qual/NonNull.html (capital N, NonNull)